hackathon_site/event/api_views.py

```diff
@@ -101,3 +106,53 @@ def post(self, request, *args, **kwargs):
         response_serializer = TeamSerializer(profile.team)
         response_data = response_serializer.data
         return Response(data=response_data, status=status.HTTP_200_OK,)
+
+
+class ProfileDetailView(mixins.UpdateModelMixin, generics.GenericAPIView):
+
+    queryset = Profile.objects.all()
+    serializer_class = ProfileSerializer
+    permission_classes = [FullDjangoModelPermissions]
+
+    @transaction.atomic
+    def patch(self, request, *args, **kwargs):
+        profile = self.get_object()
+
+        data = request.data
+        profile.id_provided = data["id_provided"]
+        profile.attended = data["attended"]
+
+        if not profile.acknowledge_rules:
+            profile.acknowledge_rules = data["acknowledge_rules"]
+        if not profile.e_signature:
+            profile.e_signature = data["e_signature"]
+
+        profile.save()
+        response_serializer = ProfileSerializer(profile)
+        response_data = response_serializer.data
+        return Response(data=response_data, status=status.HTTP_200_OK,)
+
+
+class CurrentProfileView(mixins.UpdateModelMixin, generics.GenericAPIView):
+
+    queryset = Profile.objects.all()
+    serializer_class = UserProfileSerializer
+    permission_classes = [UserHasProfile]
+
+    def get_object(self):
+        return self.request.user.profile
+
+    def patch(self, request, *args, **kwargs):
```

Comment:
Since you have `UpdateModelMixin`, you can use the `self.partial_update` method instead of doing this logic yourself:

```
def patch(self, request, *args, **kwargs):
    return self.partial_update(request, *args, **kwargs)
```

In order to do that, you'll need to move the logic into the serializer. There are two places you could do that: in the `validate` method (which returns the validated data), or in the `update` method. In this case, since we're not actually validating anything, the `update` method should be fine. 

Here's how you'd move that logic into and `update` method on the serializer (I think, this is untested):

```python
    def update(self, instance: Profile, validated_data):
        acknowledge_rules = validated_data.pop("acknowledge_rules", False)
        e_signature = validated_data.pop("e_signature", None)

        if not instance.acknowledge_rules and acknowledge_rules:
            instance.acknowledge_rules = acknowledge_rules
        if not instance.e_signature and e_signature:
            instance.e_signature = e_signature

        return super().update(instance, validated_data)
```

That's also safe against them only submitting one field or the other.